pkg/reconciler/managed/reconciler.go

```diff
@@ -693,17 +726,25 @@ func (r *Reconciler) Reconcile(ctx context.Context, req reconcile.Request) (reco
 	// not realize that the controller is still trying to reconcile
 	// (and modify or delete) the resource since they forgot to enable the
 	// feature flag.
-	if !r.managementPoliciesEnabled && (managed.GetManagementPolicy() == xpv1.ManagementObserveOnly || managed.GetManagementPolicy() == xpv1.ManagementOrphanOnDelete) {
-		log.Debug(errManagementPolicy, "policy", managed.GetManagementPolicy())
-		record.Event(managed, event.Warning(reasonManagementPolicyNotEnabled, errors.New(errManagementPolicy)))
-		managed.SetConditions(xpv1.ReconcileError(errors.New(errManagementPolicy)))
+	if isNonDefaultManagementPolicy(managementPoliciesEnabled, managed.GetManagementPolicy()) {
+		log.Debug(errManagementPolicyNonDefault, "policy", managed.GetManagementPolicy())
+		record.Event(managed, event.Warning(reasonManagementPolicyNotEnabled, errors.New(errManagementPolicyNonDefault)))
+		managed.SetConditions(xpv1.ReconcileError(errors.New(errManagementPolicyNonDefault)))
+		return reconcile.Result{}, errors.Wrap(r.client.Status().Update(ctx, managed), errUpdateManagedStatus)
+	}
+
+	if !isManagementPolicySupported(managementPoliciesEnabled, managed.GetManagementPolicy()) {
+		log.Debug(errManagementPolicyNotSupported, "policy", managed.GetManagementPolicy(), "supported",
+			supportedManagementPolicies)
```

Comment:
Not: Any reason to wrap this? I don't think its longer than other lines in the file.

Reply:
Also, what does `supportedManagementPolicies` look like when logged? Is it huge?